src/LobbyServer.hs

```diff
@@ -119,26 +119,29 @@ getGamesList remoteGames = do
   return $ map fst gameList
 
 -- |Lets a player join a 'LobbyGame'
-playerJoinGame :: Server ConcurrentClientList -> Server GamesList -> String -> Server ()
+playerJoinGame :: Server ConcurrentClientList -> Server GamesList -> String -> Server Bool
 playerJoinGame remoteClientList remoteGameList gameID = do
   clientList <- remoteClientList >>= liftIO . CC.readMVar
   gameList <- remoteGameList
   sid <- getSessionID
-  case lookupClientEntry sid clientList of
-    Just player -> liftIO $ CC.modifyMVar_ gameList $
-      \gList -> return $ addPlayerToGame player gameID gList
-
-    _ -> return ()
-
-  maybeGame <- findGameWithSid gameList
+  maybeGame <- liftIO $ findGameWithID gameID gameList
   case maybeGame of
-    Just (_,gameData) -> liftIO $ messageClients PlayerJoinedGame (players gameData)
-    nothing           -> return ()
+    Nothing           -> return False
+    Just (_,gameData) ->
+      if maxAmountOfPlayers gameData > length (players gameData) then do
+          case lookupClientEntry sid clientList of
+            Nothing     -> error "playerJoinGame: Client not registered"
+            Just player -> liftIO $ CC.modifyMVar_ gameList $
+              \gList -> return $ addPlayerToGame player gameID gList
+
+          liftIO $ messageClients PlayerJoinedGame (players gameData)
+          return True
+        else return False
 
 -- |Adds a player to a lobby game with the game ID
 addPlayerToGame :: ClientEntry -> String -> [LobbyGame] -> [LobbyGame]
 addPlayerToGame client gameID =
-  updateListElem (\(gID, GameData ps gameName) -> (gID, GameData (nub $ client:ps) gameName)) (\g -> gameID == fst g)
+  updateListElem (\(gID, gameData) -> (gID, gameData {players = nub $ client:players gameData})) (\g -> gameID == fst g)
```

Comment:
I would prefere spaces around the `:` operator.

`(\g -> gameID == fst g)` can be simplified to `(gameID ==) . fst`